service/controller/v14/resource/lifecycle/create.go

```diff
@@ -130,73 +130,67 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			n := customObject.GetNamespace()
 			o := metav1.GetOptions{}
 
-			_, err = r.g8sClient.CoreV1alpha1().NodeConfigs(n).Get(privateDNS, o)
+			_, err = r.g8sClient.CoreV1alpha1().DrainerConfigs(n).Get(privateDNS, o)
 			if errors.IsNotFound(err) {
-				r.logger.LogCtx(ctx, "level", "debug", "message", "did not find node config for guest cluster node")
+				r.logger.LogCtx(ctx, "level", "debug", "message", "did not find drainer config for guest cluster node")
 
-				err := r.createNodeConfig(ctx, customObject, *instance.InstanceId, privateDNS)
+				err := r.createDrainerConfig(ctx, customObject, *instance.InstanceId, privateDNS)
 				if err != nil {
 					return microerror.Mask(err)
 				}
 
 			} else if err != nil {
 				return microerror.Mask(err)
 			} else {
-				r.logger.LogCtx(ctx, "level", "debug", "message", "found node config for the guest cluster")
+				r.logger.LogCtx(ctx, "level", "debug", "message", "found drainer config for the guest cluster")
 
-				r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for inspection of node config for the guest cluster")
+				r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for inspection of drainer config for the guest cluster")
 			}
 		}
 	}
 
 	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", "start inspection of node configs for the guest cluster")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "start inspection of drainer configs for the guest cluster")
 
 		n := v1.NamespaceAll
 		o := metav1.ListOptions{
 			LabelSelector: fmt.Sprintf("%s=%s", key.ClusterIDLabel, key.ClusterID(customObject)),
 		}
 
-		nodeConfigs, err := r.g8sClient.CoreV1alpha1().NodeConfigs(n).List(o)
+		drainerConfigs, err := r.g8sClient.CoreV1alpha1().DrainerConfigs(n).List(o)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
-		if len(nodeConfigs.Items) > 0 {
-			for _, nodeConfig := range nodeConfigs.Items {
-				r.logger.LogCtx(ctx, "level", "debug", "message", "inspecting node config for the guest cluster")
-
-				if !nodeConfig.Status.HasFinalCondition() {
```

Comment:
The diff here is not helpful. I added the timeout condition here and restructured the conditionals.